File: TestFiles/testFilesGenerator.pl
#!/usr/bin/env perl 
#===============================================================================
#
#         FILE: testFilesGenerator.pl
#
#        USAGE: ./testFilesGenerator.pl
#
#  DESCRIPTION:
#
#      OPTIONS: ---
# REQUIREMENTS: ---
#         BUGS: ---
#        NOTES: ---
#       AUTHOR: YOUR NAME (),
# ORGANIZATION:
#      VERSION: 1.0
#      CREATED: 20/10/14 17:59:55
#     REVISION: ---
#===============================================================================

use strict;
use warnings;
use utf8;

foreach my $index ( 0 .. 3 ) {
    open( my $fh, '>:raw', "in" . 15**$index . ".bin" ) 
        or die "Unable to open: $!";
    foreach ( 0 .. 15**$index ) {
        print $fh pack('s<', int( rand(1000000) ));
    }
    close($fh);
}

File: TestFiles/testFilesGenerator.c
/*
 * =====================================================================================
 *
 *       Filename:  testFilesGenerator.c
 *
 *    Description:  
 *
 *        Version:  1.0
 *        Created:  21/10/14 21:49:17
 *       Revision:  none
 *       Compiler:  gcc
 *
 *         Author:  YOUR NAME (), 
 *   Organization:  
 *
 * =====================================================================================
 */
#include <stdio.h>
#include <stdlib.h>
#include "../arrayHandling.h"
#include "../fileHandling.h"
#include "../quickSort.h"

/* 
 * ===  FUNCTION  ======================================================================
 *         Name:  main
 *  Description:  
 * =====================================================================================
 */
  int
main ( int argc, char *argv[] )
{
  int elems[6] = {15, 25, 255, 550, 1000};
  int n;

  for ( n = 0; n < 5; n += 1 ) {
    int ranElems[elems[n]];
    char filenameIN[80] ;
    char filenameOUT[80] ;

    genRandNumbersArray(elems[n], ranElems);

    sprintf(filenameIN, "in%d.bin", elems[n]);
    sprintf(filenameOUT, "out%d.bin", elems[n]);

    writeToBinFile(filenameIN, ranElems, elems[n]);

    quickSort(ranElems, 0, elems[n]);

    writeToPlainFile(filenameOUT, ranElems, elems[n]);
  }
  return 0;
}				/* ----------  end of function main  ---------- */

